src/Shared/LayoutRenderers/AspNetRequestHeadersLayoutRenderer.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using NLog.Config;
+using NLog.LayoutRenderers;
+using NLog.Web.Internal;
+using System;
+#if !ASP_NET_CORE
+using System.Collections.Specialized;
+using System.Linq;
+#else
+using Microsoft.AspNetCore.Http;
+#endif
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET Request Headers
+    /// </summary>
+    /// <para>Example usage of ${aspnet-request-headers}</para>
+    /// <example>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-request-headers:OutputFormat=Flat}
+    /// ${aspnet-request-headers:OutputFormat=Json}
+    /// ${aspnet-request-headers:OutputFormat=Json:HeaderNames=username}
+    /// ${aspnet-request-headers:OutputFormat=Json:Exclude=access_token}
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-request-headers")]
+    [ThreadSafe]
+    public class AspNetRequestHeadersLayoutRenderer : AspNetLayoutMultiValueRendererBase
+    {
+        /// <summary>
+        /// Header names to be rendered.
+        /// If <c>null</c> or empty array, all headers will be rendered.
+        /// </summary>
+        public List<string> HeaderNames { get; set; }
+
+        /// <summary>
+        /// Gets or sets the keys to exclude from the output. If omitted, none are excluded.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+#if ASP_NET_CORE
+        public ISet<string> Exclude { get; set; }
+#else
+        public HashSet<string> Exclude { get; set; }
+#endif
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AspNetRequestHeadersLayoutRenderer" /> class.
+        /// </summary>
+        public AspNetRequestHeadersLayoutRenderer()
+        {
+            Exclude = new HashSet<string>(new[] { "ALL_HTTP", "ALL_RAW", "AUTH_PASSWORD" }, StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Renders the ASP.NET Headers appends it to the specified <see cref="StringBuilder" />.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder" /> to append the rendered data to.</param>
+        /// <param name="logEvent">Logging event.</param>
+        protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
+        {
+            var httpRequest = HttpContextAccessor.HttpContext.TryGetRequest();
+            if (httpRequest == null)
+            {
+                return;
+            }
+
+            var headers = httpRequest.Headers;
+            if (headers?.Count > 0)
+            {
+                bool excludeKeys = (HeaderNames == null || HeaderNames.Count == 0) && Exclude?.Count > 0;
```

Comment:
please rename to checkForExclude, as done here: https://github.com/NLog/NLog.Web/pull/472/commits/08ebbdd1ccabd2882717304ca142aaa9072cf805

(so also in GetHeaderValues)

Then I will merge it :)

Thanks!